src/wp-includes/option.php

```diff
@@ -918,20 +923,21 @@ function update_option( $option, $value, $autoload = null ) {
  *
  * @global wpdb $wpdb WordPress database abstraction object.
  *
- * @param string      $option     Name of the option to add. Expected to not be SQL-escaped.
- * @param mixed       $value      Optional. Option value. Must be serializable if non-scalar.
- *                                Expected to not be SQL-escaped.
- * @param string      $deprecated Optional. Description. Not used anymore.
- * @param string|bool $autoload   Optional. Whether to load the option when WordPress starts up.
- *                                Accepts 'yes'|true to enable or 'no'|false to disable.
- *                                Autoloading too many options can lead to performance problems, especially if the
- *                                options are not frequently used. For options which are accessed across several places
- *                                in the frontend, it is recommended to autoload them, by using 'yes'|true.
- *                                For options which are accessed only on few specific URLs, it is recommended
- *                                to not autoload them, by using 'no'|false. Default 'yes'.
- * @return bool True if the option was added, false otherwise.
+ * @param string      $option           Name of the option to add. Expected to not be SQL-escaped.
+ * @param mixed       $value            Optional. Option value. Must be serializable if non-scalar.
+ *                                      Expected to not be SQL-escaped.
+ * @param string      $deprecated       Optional. Description. Not used anymore.
+ * @param string|bool|null $autoload    Optional. Whether to load the option when WordPress starts up.
+ *                                      Accepts 'yes'|true to enable or 'no'|false to disable.
+ *                                      Autoloading too many options can lead to performance problems, especially if the
+ *                                      options are not frequently used. For options which are accessed across several places
+ *                                      in the frontend, it is recommended to autoload them, by using 'yes'|true.
+ *                                      For options which are accessed only on few specific URLs, it is recommended
+ *                                      to not autoload them, by using 'no'|false. Default 'yes'.
+ *
+ * @return bool                         True if the option was added, false otherwise.
  */
-function add_option( $option, $value = '', $deprecated = '', $autoload = 'yes' ) {
+function add_option( $option, $value = '', $deprecated = '', $autoload = null ) {
```

Comment:
Imho this is a bad idea, and also borders on back-compat break. All new options **have to be** set to autoload by default. 

In addition changing the function signature doesn't make sense as there is `determine_option_autoload_value()` now that can deal with oerly large options, and let plugins ov erride its automatic decision.

Reply:
See my reply in https://github.com/WordPress/wordpress-develop/pull/5671#discussion_r1419694609. Let's continue discussing there, to not have the same conversation in multiple places.